### ISSUE TITLE
Inline label editing click and save functionality

### ISSUE THREAD

**Issue by mnahmias** [adobe/aem-core-forms-components#1205]
### Expected Behaviour
When editing an inline form label clicking a point in the string should move the cursor to that point. Hitting enter should save the authored changes.

### Actual Behaviour
When editing an inline form label clicking a point in the string ends editing. Hitting enter ends editing and removes the changes.

#### Steps to Reproduce
* Create a text box field

##### Editing issue
* Select the field and click the edit button
* Click anywhere in the label to try to move the cursor
* Editing ends

##### Saving issue
* Select the field and click the edit button
* Alter the text of the label and hit enter
* Editing ends and changes are reverted

#### AEM Version (mention the exact version in case of cloud SDK)
2023.12.14697.20231215T125030Z-231200

#### AEM Forms Version
2023.10.25.00-231200

**Comment by rismehta** [comment]
@mnahmias Yes, we're aware of this issue, but it hasn't been prioritized yet as there are alternative solutions available to accomplish the same use-case
